src/reanimated2/shareables.ts

```diff
@@ -241,14 +241,28 @@ Offending code was: \`${getWorkletCode(value)}\``);
         shareableMappingCache.set(value, inaccessibleObject);
         return inaccessibleObject;
       }
-      if (__DEV__) {
-        // we freeze objects that are transformed to shareable. This should help
-        // detect issues when someone modifies data after it's been converted to
-        // shareable. Meaning that they may be doing a faulty assumption in their
+      if (__DEV__ && !isHostObject(value) && isPlainJSObject(value)) {
+        // We freeze objects that are transformed to Shareable with a meaningful warning.
+        // This should help detect issues when someone modifies data after it's been converted to
+        // Shareable. Meaning that they may be doing a faulty assumption in their
         // code expecting that the updates are going to automatically populate to
         // the object sent to the UI thread. If the user really wants some objects
-        // to be mutable they should use shared values instead.
-        Object.freeze(value);
+        // to be mutable they should use Shared Values instead.
+        Object.entries(value).forEach(([key, element]) => {
```

Comment:
Can we come up with some metric here and measure the performance impact of this change?

Reply:
Since it was development only change I neglected that but sure we can.

Reply:
That's debatable 🤓

Reply:
From the results you have provided, the difference is negligible 👍